fixieai/cli/agent/commands.py

```diff
@@ -138,71 +274,51 @@ def _update_agent_requirements(
     type=str,
     help="Additional requirements for requirements.txt. Can be specified multiple times.",
 )
-def init_agent(handle, description, entry_point, more_info_url, requirement):
+def init_agent(handle, description, entry_point, more_info_url, requirement, language):
+    templator: AgentTemplator
+    if language.lower() in ["python", "py"]:
+        templator = PythonTemplator()
+    elif language.lower() in ["typescript", "ts"]:
+        templator = TypeScriptTemplator()
+
     try:
         current_config = agent_config.load_config()
     except FileNotFoundError:
-        current_config = agent_config.AgentConfig()
+        current_config = agent_config.AgentConfig(language=language)
     current_config.handle = handle
     current_config.description = description
     current_config.entry_point = entry_point
+    current_config.language = language
     current_config.more_info_url = more_info_url
+
+    # Having this hook here is maybe a little gross.
+    templator.modify_new_agent_config(current_config)
```

Comment:
To handle `entry_point` I'd recommend:
- making the type in `AgentConfig` an `Optional[str]` and default value `None`
- removing `prompt` as you've done
- doing something along the lines of
```python
if entry_point:
   current_config.entry_point = entry_point
elif language == "python":
   current_config.entry_point = DEFAULT_PYTHON_ENTRY_POINT
elif language == "typescript":
   current_config.entry_point = DEFAULT_TYPESCRIPT_ENTRY_POINT
```

I think that probably avoids needing to pass the config through the templator